### PR TITLE
PublishedService option

### PR DESCRIPTION
**Pull request by Daverkex**
Implementation of #207

### CODE REVIEW

File: traefik/values.yaml
Comment (notsureifkevin) [inline traefik/values.yaml:48]: should we add a short description detailing what enabling this does?

File: traefik/templates/deployment.yaml
Comment (notsureifkevin) [inline traefik/templates/deployment.yaml:122]: I suspect that if the user disables the `kubernetesIngress` provider, but leaves the `publishService` enabled it will result in a misconfigured traefik that will not start.

should we consider adding a condition check for `kubernetesIngress.enabled` here as well?

File: traefik/values.yaml
Comment (Daverkex) [reply traefik/values.yaml:49]: Please feel free to reopen if anything is not correct.

File: traefik/templates/deployment.yaml
Comment (Daverkex) [reply traefik/templates/deployment.yaml]: Hi @notsureifkevin,
Thank you for the review. I tested this case and nothing happened because all `publishedService` is inside the if of `kubernetesIngress enabled`. But I discover a misconfiguration if `service.enabled` is false and I added an extra check.

Is it resolved for you?

File: traefik/Chart.yaml
Comment (SantoDE) [inline traefik/Chart.yaml:4]: will need a rebase soon, once the PR #221 is merged to match the appVersion. You can set it now already

Comment (Daverkex) [reply traefik/Chart.yaml:4]: Hi @SantoDE, this PR is older and is up to date after that #221. I think that the logical is to integrate this PR first and rebase in the other.

Comment (SantoDE) [reply traefik/Chart.yaml:4]: #221 is now merged, so please one last rebase :-)

**Comment by monotek** [comment]
Whats missing here?
I need ths feature too :D

@Daverkex 
Could you please also update the traefik docker image to the most recent version?
It's 2.2.8 now.

**Comment by Daverkex** [comment]
Hi @monotek 
It's done ;) I'm waiting for maintainers review, but this repo moves too slow. Maybe they are on holiday.

**Comment by monotek** [comment]
@Daverkex 

Thanks for the update :)

I thought a LGTM or something is missing...

**Comment by SantoDE** [comment]
Yes, I was on vacation :-) I'll work through the repository throughout this week 👍

**Comment by monotek** [comment]
@SantoDE 

Many thanks! :)

Hope you spend some wonderful days of rest :)

**Comment by SantoDE** [comment]
@Daverkex can you fix the conflict? I'll merge this right after :)

**Comment by Daverkex** [comment]
This is the third time that this PR needs to fix conflicts. Because others PRs newer are prioritized instead of this.
No other issues to stop this PR.

What's the criterion to select the next PR to integrate?

This is my last update in this PR to resolve conflicts.
Feel free to resolve conflict or close the PR.

**Comment by SantoDE** [comment]
Hello @Daverkex,

I did resolve the conflicts for you.

The only criterion that did hit you was "time". I try to keep the PR's active currently so I ping and check the PR's open. Once a PR is ready to get merged, we'll merge it. So in short: there were just PR's faster than this one.

I did fix the conflicts on this PR now and will merge it. Thanks for your work 🎉

**Review by SantoDE — APPROVED** [review]
LGTM

**Pull request merged**
